torch/csrc/jit/script/script_type_parser.cpp

```diff
@@ -162,6 +162,39 @@ c10::optional<std::string> parseBaseTypeName(const Expr& expr) {
   return at::nullopt;
 }
 
+// Parse NamedTuple following the grammar like below:
+// NamedTuple('Employee', [('name', str), ('id', int)])
+// See also: https://docs.python.org/3/library/typing.html#typing.NamedTuple
+TypePtr parseNamedTuple(const Expr& expr) {
+  auto apply = Apply(expr);
+  auto callee = apply.callee();
+  if (callee.kind() != TK_VAR || Var(callee).name().name() != "NamedTuple") {
```

Comment:
Will this work if I `import collections` and then say `collections.NamedTuple(...)`?

Reply:
(I'm not too familiar with this corner of the JIT, but I don't see any name resolution machinery so I suspect the answer is no...)

Reply:
I guess not. The name resolution in JIT is working i a weird way, for example:
```python
import torch as t


@t.jit.script
def f(x):
    # type: (torch.Tensor) -> int
    # this works, but who is torch?
    return 1

@t.jit.script
def f(x):
    # type: (t.Tensor) -> int
    # this does not work...
    return 1
```

Reply:
And also: https://github.com/pytorch/pytorch/issues/16492

Reply:
Yes, probably this can't be fixed properly until torchscript gets a real name resolution pass.